src/server/util/kubectl.cc

```diff
@@ -93,6 +116,27 @@ static std::string generate_global_object(
   std::string signature =
       SignatureToString(object["signature"].get<Signature>());
   std::string type_name = object["typename"].get_ref<std::string const&>();
+
+  std::string job_name = "none";
+  if (object.contains("JOB_NAME")) {
+    job_name = object["JOB_NAME"].get_ref<std::string const&>();
+  } else {
+    std::cout << "Environment variable JOB_NAME not set" << std::endl;
+  }
+  std::string client_pod_name = "none-set";
+  if (object.contains("POD_NAME")) {
+    client_pod_name = object["POD_NAME"].get_ref<std::string const&>();
+  } else {
+    std::cout << "Environment variable POD_NAME not set" << std::endl;
+  }
+  std::string client_pod_namespace = "default";
+  if (object.contains("POD_NAMESPACE")) {
+    client_pod_namespace =
+        object["POD_NAMESPACE"].get_ref<std::string const&>();
+  } else {
+    std::cout << "Environment variable POD_NAMESPACE not set" << std::endl;
+  }
```

Comment:
Use `object.value("POD_NAME", "")`, etc. to simplify those pieces, using an empty string as default value should be OK.

Don't use `std::cout` to logginng in vineyardd. We have already introduced glog into our dependencies.

Reply:
Make sense to me.